### PR TITLE
Drop python 3.8 support

### PR DESCRIPTION
**Pull request by enadeau**
Remove supports for python 3.8 following discussion on #301

## Description

- Remove python 3.8 support and ci run against that versions
- run pyupgrade for python 3.9+

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [x] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

### CODE REVIEW

**Comment by lwasser** [comment]
@enadeau this is cool - how does pyupgrade work? i just looked at the readme. do you tell it `pyupgrade python-version` or something to that effect and it just fixes everything?

**Comment by lwasser** [comment]
ok i did just a bit more digging. 

I think we should release a minor version based on the existing new things BEFORE this merge. Then we can create a new version with this drop of 3.8. this will allow anyone running python 3.8 a bit of time to pin their workflow to stravalib 1.3.4 which still supports 3.8. Then maybe this is a 1.4 bump - i guess it's breaking in theory for anyone running Python 3.8. but i'm not sure if we want to jump to 2.0 - i was thinking 2.0 might be the pydantic 2.0 upgrade?

i'm not sure if @jsamoocha has time to weight in on this - but just want to think this through prior to merging something that will break some peoples applications if they are running 3.8.

**Comment by lwasser** [comment]
One other thought @enadeau  - i'm talking with folks and thinking about our process here. If we make the changes in this PR it will REALLY break people's code running python 3.8. I wonder if we always want to wait on big code updates like this one UNTIL 3.8 is EOL but we could still drop support for 3.8 in our builds and such. 

OR is the issue with pint such that it requires code that is >=3.9? 
thank you. i'm just trying to carefully think this through in case there is a way that we can accomodate some 3.8 users for the time being in a more undocumented way (we say we support 3.9 + but we accomodate 3.8 IF that is possible. thoughts?

I'm happy to make a release regardless of everything except this pr in the upcoming days!

**Comment by jsamoocha** [comment]
Having a minor release with the typing changes, then going to 1.4 on the drop of Python 3.8, and then going to 2.0 whenever we drop support for the pre-pydantic API seems a totally reasonable approach, @lwasser.

**Comment by lwasser** [comment]
great! i'm fine with a minor bump for what's been merged to date. My only concern is allowing people to pin to the most recent version with bugs fixed if they need to. 

so to proceed:

1. i'll create a 1.4 release with what's been merged on main. 
2. Then we can drop support for 3.8 here, rebase to fix #415  and create another minor release - fully typed with those two additional pr's. 1.5 
3. After that i'll work a bit more on the nox migration (moving stravalib to a src/ layout and adding a nox build session) and updating docs with NEP-29 / python version support details (and then we can close #254 as well!  🎉 

i feel good about it given the consensus we have here at this point. let me know if i'm missing a step. i'll create a new issue to track all of the steps here as well :)

File: pyproject.toml
Comment (lwasser) [inline pyproject.toml:28]:
```suggestion
     {name = "Yihong"},
     {name = "Émile Nadeau"},
    ]
```

@enadeau adding you here as a maintainer.

**Review by lwasser — APPROVED** [review]
This PR looks good to me! i made a small change adding you to the maintainer list in the pyproject.toml file, @enadeau but otherwise it's good and i'll merge.

Would you like to create the next release to make sure the steps are clear / you have correct access? or would like prefer that I do this one. it's up to you!!

**Comment by lwasser** [comment]
Everything is running here. i'm going to merge this y'all. then @enadeau you can rebase and your other PR should build properly.

**Pull request merged**
